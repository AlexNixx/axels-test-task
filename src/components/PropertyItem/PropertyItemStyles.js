import {Card} from "react-bootstrap";
import {Link} from "react-router-dom";
import styled from "styled-components";

export const StyledLink = styled(Link)`
  text-decoration: none;
`

export const StyledCard = styled(Card)`
  width: 18rem;
  height: 25rem;
  overflow: hidden;
  cursor: pointer;
  
`

export const StyledImage = styled(Card.Img)`
  width: 100%;
  height: 15rem;
  object-fit: cover;
  transition: transform 0.25s;

  &:hover {
    transform: scale(1.05);
  }
`