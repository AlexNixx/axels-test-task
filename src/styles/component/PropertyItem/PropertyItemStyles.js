import { Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

export const StyledLink = styled(Link)`
    text-decoration: none;
`;

export const StyledCard = styled(Card)`
    height: 20rem;
    overflow: hidden;
    cursor: pointer;
`;

export const StyledImage = styled(Card.Img)`
    height: 10rem;
    width: 100%;
    object-fit: cover;
    transition: transform 0.25s;

    &:hover {
        transform: scale(1.05);
    }
`;
