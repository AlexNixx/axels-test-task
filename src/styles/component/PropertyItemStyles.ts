import { Card } from 'react-bootstrap';
import styled from 'styled-components';

export const StyledCard = styled(Card)`
    height: 20rem;
    overflow: hidden;
`;

export const StyledImage = styled(Card.Img)`
    height: 10rem;
    width: 100%;
    object-fit: cover;
    transition: transform 0.25s;

    &:hover {
        transform: scale(1.05);
    }
`;
