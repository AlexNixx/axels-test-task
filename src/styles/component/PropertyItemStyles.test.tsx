import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { StyledCard, StyledImage } from './PropertyItemStyles';

describe('Test PropertyItemStyles', () => {
    test('Check StyledCart styles', () => {
        const tree = renderer.create(<StyledCard />).toJSON();
        expect(tree).toHaveStyleRule('height', '20rem');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toMatchSnapshot();
    });

    test('Check StyledImage styles', () => {
        const tree = renderer.create(<StyledImage />).toJSON();
        expect(tree).toHaveStyleRule('height', '10rem');
        expect(tree).toHaveStyleRule('object-fit', 'cover');
        expect(tree).toHaveStyleRule('transform', 'scale(1.05)', {
            modifier: ':hover'
        });
        expect(tree).toMatchSnapshot();
    });
});
