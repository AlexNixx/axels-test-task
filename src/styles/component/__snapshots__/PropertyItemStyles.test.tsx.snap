// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PropertyItemStyles Check StyledCart styles 1`] = `
.c0 {
  height: 20rem;
  overflow: hidden;
}

<div
  className="c0 card"
/>
`;

exports[`Test PropertyItemStyles Check StyledImage styles 1`] = `
.c0 {
  height: 10rem;
  width: 100%;
  object-fit: cover;
  transition: transform 0.25s;
}

.c0:hover {
  transform: scale(1.05);
}

<img
  className="card-img c0"
/>
`;
